specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json

```diff
@@ -0,0 +1,1084 @@
+          ],
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "description": "The list of user identities associated with the resource. The user identity dictionary key references will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'.",
+          "additionalProperties": {
+            "$ref": "#/definitions/UserAssignedIdentitiesProperties"
+          }
+        }
+      }
+    },
+    "EncryptionProperties": {
+      "description": "The encryption settings for automation account",
+      "properties": {
+        "keyVaultProperties": {
+          "$ref": "#/definitions/KeyVaultProperties",
+          "description": "Key vault properties."
+        },
+        "keySource": {
+          "type": "string",
+          "description": "Encryption Key Source",
+          "enum": [
+            "Microsoft.Automation",
+            "Microsoft.Keyvault"
+          ],
+          "x-ms-enum": {
+            "name": "EncryptionKeySourceType",
+            "modelAsString": false
+          }
+        },
+        "identity": {
+          "description": "User identity used for CMK.",
+          "properties": {
+            "userAssignedIdentity": {
+              "type": "object",
+              "description": "The user identity used for CMK. It will be an ARM resource id in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'."
+            }
+          }
+        }
+      }
+    },
+    "KeyVaultProperties": {
+      "type": "object",
+      "description": "Settings concerning key vault encryption for a configuration store.",
+      "properties": {
+        "keyvaultUri": {
+          "description": "The URI of the key vault key used to encrypt data.",
+          "type": "string"
+        },
+        "keyName": {
+          "description": "The name of key used to encrypt data.",
+          "type": "string"
+        },
+        "keyVersion": {
+          "description": "The key version of the key used to encrypt data.",
+          "type": "string"
+        }
+      }
+    },
+    "DeletedRunbookProperties": {
+      "type": "object",
+      "properties": {
+        "runbookId": {
+          "type": "string",
+          "description": "Gets or sets the Runbook Id."
+        },
+        "runbookType": {
+          "type": "string",
+          "description": "Type of the runbook."
+        },
+        "runtime": {
+          "type": "string",
+          "description": "Gets or sets runtime of the runbook."
+        },
+        "runtimeEnvironment": {
+          "type": "string",
+          "description": "Environment of the runbook.",
+          "x-nullable": true
```

Comment:
>"x-nullable": true [](http://example.com/codeflow?start=10&length=18)

strings don't usually need x-nullable, so I'm wondering if there's some special reason again.

if you want to distinguish strings that are always present from ones which are not, 'required' could be the way to do that.

Reply:
but if its already shipped its probably a breaking change so never mind..

Reply:
Got its, Yes its already shipped